system/security/shared/caching.cpp

```diff
@@ -210,7 +210,10 @@ void CResPermissionsCache::remove(SecResourceType rtype, const char* resourcenam
 
 CPermissionsCache::~CPermissionsCache()
 {
-    g_mapCache.clear();
+    {
+        CriticalBlock block(PCCritSect);
```

Comment:
The map.erase() function is not deleting the CPermissionCache instance, only clearing up the map.  We are already in the DTOR for CPermissionCache, which means the last reference to this object has just been released.  As an aside, the g_mapCache is only used at program initialization to distribute references to the cache, so for both reasons we are good